tests/dort_test.py

```diff
@@ -0,0 +1,202 @@
+import copy
+import inspect
+import itertools
+import sys
+import unittest
+import warnings
+from typing import Optional
+
+import onnxruntime
+import torch
+import torch.nn as nn
+import torch.onnx
+from onnx.inliner import inline_local_functions
+
+
+def make_aot_ort(
+    dynamic: bool = False,
+    ort_optimization_level: Optional[str] = None,
+) -> tuple:
+    code = inspect.getsource(
+        torch.onnx._internal.onnxruntime  # pylint: disable=protected-access
+    )
+    if "optimizer.optimize" not in code:
+        raise unittest.SkipTest(
+            f"torch=={torch.__version__!r} is not recent enough, "
+            f"file {torch.onnx._internal.onnxruntime.__file__!r} "  # pylint: disable=protected-access
+            f"does not optimize the exported model."
+        )
+
+    ort_session_options = onnxruntime.SessionOptions()
+    if ort_optimization_level is not None:
+        assert hasattr(onnxruntime.GraphOptimizationLevel, ort_optimization_level), (
+            f"Unexpected value {ort_optimization_level!r} for GraphOptimizationLevel, "
+            f"expecting one of the values in {dir(onnxruntime.GraphOptimizationLevel)}"
+        )
+        ort_session_options.graph_optimization_level = getattr(
+            onnxruntime.GraphOptimizationLevel, ort_optimization_level
+        )
+
+    export_options = torch.onnx.ExportOptions(dynamic_shapes=dynamic)
+
+    def inline_function(*args, **kwargs):  # pylint: disable=unused-argument
+        first_model_proto = args[0]
+
+        next_model = inline_local_functions(first_model_proto)
+
+        del first_model_proto.graph.node[:]
+        del first_model_proto.functions[:]
+        del first_model_proto.graph.initializer[:]
+        del first_model_proto.opset_import[:]
+        first_model_proto.graph.node.extend(next_model.graph.node)
+        first_model_proto.functions.extend(next_model.functions)
+        first_model_proto.graph.initializer.extend(next_model.graph.initializer)
+        first_model_proto.opset_import.extend(next_model.opset_import)
+
+        return first_model_proto
```

Comment:
Avoid nested declarations of functions if it doesn't close over local variables?